connexion/apis/flask_api.py

```diff
@@ -130,75 +120,13 @@ def _build_response(
         return flask.current_app.response_class(**kwargs)
 
     @classmethod
-    def _serialize_data(cls, data, mimetype):
-        if isinstance(mimetype, str) and is_json_mimetype(mimetype):
-            body = cls.jsonifier.dumps(data)
-        elif not (isinstance(data, bytes) or isinstance(data, str)):
-            warnings.warn(
-                "Implicit (flask) JSON serialization will change in the next major version. "
-                "This is triggered because a response body is being serialized as JSON "
-                "even though the mimetype is not a JSON type. "
-                "This will be replaced by something that is mimetype-specific and may "
-                "raise an error instead of silently converting everything to JSON. "
-                "Please make sure to specify media/mime types in your specs.",
-                FutureWarning,  # a Deprecation targeted at application users.
-            )
-            body = cls.jsonifier.dumps(data)
-        else:
-            body = data
-
-        return body, mimetype
-
-    @classmethod
-    def get_request(cls, *args, **params):
-        # type: (*Any, **Any) -> ConnexionRequest
-        """Gets ConnexionRequest instance for the operation handler
-        result. Status Code and Headers for response.  If only body
-        data is returned by the endpoint function, then the status
-        code will be set to 200 and no headers will be added.
-
-        If the returned object is a flask.Response then it will just
-        pass the information needed to recreate it.
-
-        :rtype: ConnexionRequest
-        """
-        flask_request = flask.request
-        scope = flask_request.environ["asgi.scope"]
-        context_dict = scope.get("extensions", {}).get("connexion_context", {})
-        setattr(flask.globals.request_ctx, "connexion_context", context_dict)
-        request = ConnexionRequest(
-            flask_request.url,
-            flask_request.method,
-            headers=flask_request.headers,
-            form=flask_request.form,
-            query=flask_request.args,
-            body=flask_request.get_data(),
-            json_getter=lambda: flask_request.get_json(silent=True),
-            files=flask_request.files,
-            path_params=params,
-            context=context_dict,
-            cookies=flask_request.cookies,
-        )
-        logger.debug(
-            "Getting data and status code",
-            extra={
-                "data": request.body,
-                "data_type": type(request.body),
-                "url": request.url,
-            },
-        )
-        return request
+    def get_request(cls, **kwargs) -> ConnexionRequest:
+        uri_parser = kwargs.pop("uri_parser")
+        return ConnexionRequest(flask.request, uri_parser=uri_parser)
```

Comment:
I simplified the ConnexionRequest since I don't believe we should still provide such a high abstraction layer here anymore. This should only be implemented for Flask and Starlette. Other frameworks are supported via the ASGI interface.